File: Backend/fantastic_food/domain/src/main/java/com/sander/fantastic_food/Recipe.java
package com.sander.fantastic_food;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class Recipe {
    private int id;
    private String recipeName;
    private String instructions;
    private int preparationTime;
    private String difficulty;
    private String recipeImage;
    private List<Allergy> recipeAllergies;
    private List<MealPlan> recipeMealPlans;


    @Override
    public boolean equals(Object obj) {
        Recipe other = (Recipe) obj;
        return this.getId() == other.getId();
    }
}

File: Flutter/fantastic_food/lib/domain/recipe.dart
import 'package:fantastic_food/domain/allergy.dart';
import 'package:fantastic_food/domain/meal_plan.dart';

class Recipe {
  int id;
  String recipeName;
  String instructions;
  int preparationTime;
  String difficulty;
  String recipeImage;
  List<Allergy> recipeAllergies;
  List<MealPlan> recipeMealPlans;

  Recipe(this.id, this.recipeName, this.instructions, this.preparationTime, this.difficulty,
    this.recipeImage, this.recipeAllergies, this.recipeMealPlans);

  Recipe.clear();

  factory Recipe.fromJson(dynamic json) {
    return Recipe(json['id'] as int, json['recipeName'] as String, json['instructions'] as String, 
        json['preparationTime'] as int, json['difficulty'] as String, json['recipeImage'] as String,
        List<Allergy>.from(json['recipeAllergies'].map((model)=> Allergy.fromJson(model))),
        List<MealPlan>.from(json['recipeMealPlans'].map((model)=> MealPlan.fromJson(model))));
  }

   Map<String, dynamic> toJson() {
    return {
      "id": this.id,
      "recipeName": this.recipeName,
      "instructions": this.instructions,
      "preparationTime": this.preparationTime,
      "difficulty": this.difficulty,
      "recipeImage": this.recipeImage,
      "recipeAllergies": this.recipeAllergies,
      "recipeMealPlans": this.recipeMealPlans
    };
  }

  @override
  bool operator ==(Object other) {
    return other is Recipe && other.id == id;
  }

  int get hashcode => id.hashCode^recipeName.hashCode;
}